intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/indexing/indexing.py

```diff
@@ -1,211 +1,212 @@
-            vectordb = Chroma.from_documents(documents=documents, embedding=self.embeddings,
-                                             persist_directory=self.persist_dir)
-            vectordb.persist()
-            print("The local knowledge base has been successfully built!")
-            return vectordb
-        elif self.retrieval_type == "sparse":
-            print("Will be removed in another PR")
-
-
-    def KB_append(self, input):  ### inmemory documentstore please use KB construct
-        if isinstance(input, str):
-            if os.path.isfile(input):
-                data_collection = self.parse_document(input)
-            elif os.path.isdir(input):
-                data_collection = self.batch_parse_document(input)
-            else:
-                print("Please check your upload file and try again!")
-        elif isinstance(input, List):
-            try:
-                data_collection = self.parse_html(input)
-            except:
-                print("The given link/str is unavailable. Please try another one!")
-        else:
-            print("The input format is invalid!")
-
-        if self.retrieval_type == "dense":
-            documents = []
-            for data, meta in data_collection:
-                if len(data) < 5:
-                    continue
-                metadata = {"source": meta}
-                new_doc = Document(page_content=data, metadata=metadata)
-                documents.append(new_doc)
-            assert documents != [], "The given file/files cannot be loaded."
-            vectordb = Chroma.from_documents(documents=documents, embedding=self.embeddings,
-                                             persist_directory=self.persist_dir)
-            vectordb.persist()
-            print("The local knowledge base has been successfully built!")
-            return Chroma(persist_directory=self.persist_dir, embedding_function=self.embeddings)
-        elif self.retrieval_type == "sparse":
-            print("Will be removed in another PR.")
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Wrapper for parsing the uploaded user file and then make document indexing."""
+
+
+import os, re
+from typing import Dict, List
+from langchain.vectorstores.chroma import Chroma
+from langchain.docstore.document import Document
+from langchain.embeddings import GooglePalmEmbeddings
```

Comment:
move to embedding?

Reply:
`GooglePalmEmbeddings` is not for huggingface models.